sdk/src/http/restricted.rs

```diff
@@ -0,0 +1,404 @@
+        self.allowed_hosts.as_deref()
+    }
+}
+
+impl<T: SyncHttpResolver> SyncHttpResolver for RestrictedResolver<T> {
+    fn http_resolve(
+        &self,
+        request: Request<Vec<u8>>,
+    ) -> Result<Response<Box<dyn Read>>, HttpResolverError> {
+        if self
+            .allowed_hosts()
+            .is_none_or(|hosts| is_uri_allowed(hosts, request.uri()))
+        {
+            self.inner.http_resolve(request)
+        } else {
+            Err(HttpResolverError::UriDisallowed {
+                uri: request.uri().to_string(),
+            })
+        }
+    }
+}
+
+#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
+impl<T: AsyncHttpResolver + Sync> AsyncHttpResolver for RestrictedResolver<T> {
+    async fn http_resolve_async(
+        &self,
+        request: Request<Vec<u8>>,
+    ) -> Result<Response<Box<dyn Read>>, HttpResolverError> {
+        if self
+            .allowed_hosts()
+            .is_none_or(|hosts| is_uri_allowed(hosts, request.uri()))
+        {
+            self.inner.http_resolve_async(request).await
+        } else {
+            Err(HttpResolverError::UriDisallowed {
+                uri: request.uri().to_string(),
+            })
+        }
+    }
+}
+
+/// A host/scheme pattern used to restrict network requests.
+///
+/// Each pattern may include:
+/// - A scheme (e.g. `https://` or `http://`)
+/// - A hostname, which may have a single leading wildcard (e.g. `*.contentauthenticity.org`)
+///
+/// Matching is case-insensitive. A wildcard pattern such as `*.contentauthenticity.org` matches
+/// `sub.contentauthenticity.org`, but does not match `contentauthenticity.org` or `fakecontentauthenticity.org`.
+/// If a scheme is present in the pattern, only URIs using the same scheme are considered a match. If the scheme
+/// is omitted, any scheme is allowed as long as the host matches.
+#[cfg_attr(
+    feature = "json_schema",
+    derive(schemars::JsonSchema),
+    schemars(with = "String")
+)]
+#[derive(Debug, Clone, PartialEq)]
+pub struct HostPattern {
+    pattern: String,
+    scheme: Option<String>,
+    host: Option<String>,
+}
+
+impl HostPattern {
+    /// Creates a new `HostPattern` with the given pattern.
+    pub fn new(pattern: &str) -> Self {
+        let pattern = pattern.to_ascii_lowercase();
+        let (scheme, host): (Option<String>, Option<String>) =
+            if let Some(host) = pattern.strip_prefix("https://") {
+                (Some("https".to_owned()), Some(host.to_owned()))
+            } else if let Some(host) = pattern.strip_prefix("http://") {
+                (Some("http".to_owned()), Some(host.to_owned()))
+            } else {
+                (None, Some(pattern.clone()))
+            };
+
+        Self {
+            pattern,
+            scheme,
+            host,
+        }
+    }
+
+    /// Returns if the given URI matches the `HostPattern`.
+    pub fn matches(&self, uri: &Uri) -> bool {
```

Comment:
What about IPs v4? Should those be allowed? Can you add a test with `https://127.0.0.1` for instance? Same question for IPv6.

Asking because there might be IPs used directly in development settings. Or maybe we should disallow IPs. To be clarified, @scouten-adobe @gpeacock @ok-nick.

Reply:
I'd prefer to allow explicit IP addresses. I could imagine some environments where that might be required.